dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -455,6 +460,370 @@ def traverse(mtree_node):
+                                      dtypes.ScheduleType.GPU_Device):
+                return False
+
+        candidates = InlineTransients._candidates(sdfg, graph, nsdfg)
+        return len(candidates) > 0
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        return graph.label
+
+    def apply(self, sdfg):
+        state: SDFGState = sdfg.nodes()[self.state_id]
+        nsdfg_node: nodes.NestedSDFG = self.nsdfg(sdfg)
+        nsdfg: SDFG = nsdfg_node.sdfg
+        toremove = InlineTransients._candidates(sdfg, state, nsdfg_node)
+
+        for dname, cname in toremove.items():
+            # Make nested SDFG data descriptors transient
+            nsdfg.arrays[cname].transient = True
+
+            # Remove connectors from node
+            nsdfg_node.remove_in_connector(cname)
+            nsdfg_node.remove_out_connector(cname)
+
+            # Remove data descriptor from outer SDFG
+            del sdfg.arrays[dname]
+
+        # Remove edges from outer SDFG
+        for e in state.in_edges(nsdfg_node):
+            if e.data.data not in toremove:
+                continue
+            tree = state.memlet_tree(e)
+            for te in tree:
+                state.remove_edge_and_connectors(te)
+            # Remove newly isolated node
+            state.remove_node(tree.root().edge.src)
+
+        for e in state.out_edges(nsdfg_node):
+            if e.data.data not in toremove:
+                continue
+            tree = state.memlet_tree(e)
+            for te in tree:
+                state.remove_edge_and_connectors(te)
+            # Remove newly isolated node
+            state.remove_node(tree.root().edge.dst)
+
+
+class ASTRefiner(ast.NodeTransformer):
+    def __init__(self,
+                 to_refine: str,
+                 refine_subset: subsets.Subset,
+                 sdfg: SDFG,
+                 indices: Set[int] = None) -> None:
+        self.to_refine = to_refine
+        self.subset = refine_subset
+        self.sdfg = sdfg
+        self.indices = indices
+
+    def visit_Subscript(self, node: ast.Subscript) -> ast.Subscript:
+        if astutils.rname(node.value) == self.to_refine:
+            rng = subsets.Range(
+                astutils.subscript_to_slice(node,
+                                            self.sdfg.arrays,
+                                            without_array=True))
+            rng.offset(self.subset, True, self.indices)
+            return ast.copy_location(
+                astutils.slice_to_subscript(self.to_refine, rng), node)
+
+        return self.generic_visit(node)
+
+
+@registry.autoregister_params(singlestate=True)
+@make_properties
+class RefineNestedAccess(transformation.Transformation):
+    """ 
+    Modifies memlets of a nested SDFG such that the outer subsets are smallest
+    and the inner subsets are offsetted to zero. This helps with subsequent
+    transformations on the outer SDFGs.
```

Comment:
I don't understand what this does, can you give an example of what it does and why this is better?